### PR TITLE
[FEATURE] adding helm release action

### PR DESCRIPTION
**Pull request by nicolastakashi**
# Description

Adding Github Action to publish helm chart on Github Registry

# Screenshots

N/A

# Checklist

- [X] Pull request has a descriptive title and context useful to a reviewer.
- [X] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [X] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [ ] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [ ] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [ ] Visual tests are stable and unlikely to be flaky. See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests) and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

**Review by celian-garcia — COMMENTED** [review]
Not really knownledgeable on this type of gh action/releasing. I would wait for the @Nexucis opinion

File: .github/workflows/helm-release.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v3.5
+        with:
+          version: v3.12.0
```
Comment (Nexucis) [inline .github/workflows/helm-release.yaml:33]: perhaps we could leave it empty to always get the latest Helm version

Comment (Nexucis) [reply .github/workflows/helm-release.yaml:33]: btw to be coherent we should align the version with the linter workflow: https://github.com/perses/perses/blob/main/.github/actions/setup_helmlint/action.yaml#L14

So either upgrade the version, or remove it so we can get the latest one everytime

**Comment by Nexucis** [comment]
Thank you @nicolastakashi for taking care about this part !

It looks good to me overall. Once the versions are aligned we should be good to merge this one.
Perhaps we should also add a README to the helm chart. I'm wondering if the `index.html` is generated from the README

**Comment by nicolastakashi** [comment]
@Nexucis I'm thinking about build readme after we push it so I can test it and add the proper registry address, wdyt?

File: charts/perses/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+To install the chart with the release name `perses`:
+
+```
+helm repo add perses https://perses.github.io/perses
+helm repo update
+helm install perses perses
+```
+
+## Values
```
Comment (Nexucis) [inline charts/perses/README.md:19]: I'm partially convinced about this way but why not.

Would be cool to have the description so we don't automatically need to look at the file `values.yaml` to know the meaning of the key.

File: .github/workflows/helm-lint-and-test.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        with:
+          fetch-depth: 0
+      - name: setup
+        uses: ./.github/actions/setup_helmlint
+        with:
+          create-kind-cluster: "true"
+      - name: run chart-testing (install)
+        run: ct install --charts charts/perses
+      - name: install mdox
+        run: go install github.com/bwplotka/mdox@latest
```
Comment (Nexucis) [inline .github/workflows/helm-lint-and-test.yaml:25]: I don't get how `go` has been installed and I'm wondering if it is the explanation about why `mdox` is not found

File: .github/workflows/helm-lint-and-test.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      - name: run chart README lint
+        run: make --always-make update-helm-readme && make --always-make fmt-docs && make --always-make checkdocs
```
Comment (Nexucis) [inline .github/workflows/helm-lint-and-test.yaml:27]: running simply `make` without the option `always-make` should work don't you think ?

Comment (nicolastakashi) [reply .github/workflows/helm-lint-and-test.yaml]: Ops!
It was a copy and paste from my terminal hehe

File: .github/workflows/helm-lint-and-test.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+          enable_go: true
+
+      - name: run chart-testing (install)
+        run: ct install --charts charts/perses
+
+      - name: install mdox
+        run: go install github.com/bwplotka/mdox@latest
+
+      - name: run chart README lint
+        run: make update-helm-readme && make fmt-docs && make checkdocs
```
Comment (Nexucis) [inline .github/workflows/helm-lint-and-test.yaml:36]: since `make update-helm-readme` is already running `fmt-docs` no need to do it again ;)

**Review by Nexucis — APPROVED** [review]
nice, thanks @nicolastakashi !

**Pull request merged**